torch/distributed/_functional_collectives.py

```diff
@@ -580,6 +634,18 @@ def mk_out_tensor(input):
 
     return [mk_out_tensor(t) for t in inputs]
 
+def _batch_isend_irecv_meta(p2p_op_list, *args):
```

Comment:
I'm not super familiar with the functionality here, but technically if we are only creating tensors for recv operations the number of new `out_tensors` is most likely wrong. It might be easier to address once we implement inductor support.